src/main/scala/cromwell/binding/values/WdlFile.scala

```diff
@@ -9,9 +9,19 @@ object WdlFile {
     if (path1.endsWith("/") || path2.startsWith("/")) path1 + path2
     else path1 + "/" + path2
   }
+
+  def apply(value: String, isGlob: Boolean = false): WdlFile = {
+    if(isGlob) {
+      WdlGlobFile(value)
+    } else {
+      WdlSingleFile(value)
+    }
+  }
 }
 
-case class WdlFile(value: String) extends WdlPrimitive {
+sealed trait WdlFile extends WdlPrimitive {
+  val value: String
+  val isGlob: Boolean
```

Comment:
This is a six-of-one-half-dozen-of-another type of thing, I'm not even sure which pattern I tend to prefer but I like to throw them out there w/ these ADTs but it's also possible to handle the `isGlob` attribute in a more FP style than the classic OO by doing a pattern match function in the trait:

    def isGlob = this match {
      case x: WdlSingleFile = false
      case z: WdlGlobFile = true
    }

since it's a sealed trait we have exhaustiveness testing so you're required to define it for everything.

Pro: You can see the logic for all of the individual sum types in one place
Con: You have to remember to modify the base trait every type you add to the sum type